### PR TITLE
Add timer card and badge

### PR DESCRIPTION
**Pull request by balloob**
This adds a timer domain icon, badge and state card.

Badge: image

State card:
image

Badge in action: https://youtu.be/me89eN4LxOQ

CC @Danielperna84

### CODE REVIEW

**Comment by andrey-git** [comment]
Suggestion: Start times on connect and cancel them on disconnect.

**Comment by balloob** [comment]
But then we would run an interval when the timer is idle too?

The timer component is pretty robust as in it _shouldn't_ go from active -> active. Remaining is only updated when it transitions to active.

**Comment by andrey-git** [comment]
I meant that on disconnect the timer should stop and on re-connect it can resume, so it won't do any work after I implement disconnect-in-invisible-tab.

**Comment by balloob** [comment]
Doh - fixed. Added some tests too.

File: js/common/util/seconds_to_duration.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export default function secondsToDuration(d) {
+  const h = Math.floor(d / 3600);
+  const m = Math.floor((d % 3600) / 60);
+  const s = Math.floor(d % 3600 % 60);
+
+  if (h > 0) {
+    return `${h}:${String(m).padStart(2, 0)}:${String(s).padStart(2, 0)}`;
```
Comment (balloob) [inline js/common/util/seconds_to_duration.js]: [Browser Compatability](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart#Browser_compatibility) for String.prototype.padStart.

image

Comment (balloob) [reply js/common/util/seconds_to_duration.js]: Nah, just realized that this is way too aggressive for what we currently have as a min version:

image

File: src/state-summary/state-card-timer.html [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+        margin-left: 16px;
+        text-align: right;
+        line-height: 40px;
+      }
+    </style>
+
+    <div class='horizontal justified layout'>
+      <state-info state-obj="[[stateObj]]" in-dialog='[[inDialog]]'></state-info>
+      <div class='state'>[[secondsToDuration(timeRemaining)]] <i>([[stateObj.state]])</i></div>
```
Comment (armills) [inline src/state-summary/state-card-timer.html]: Just an opinion, but I think the active/inactive text here isn't necessary. It's already pretty clear when a timer is active from the icon and the fact that the time is changing. IMO simpler is better.

If we do want to keep it we should pass it through `computeStateDisplay` for localization.

Comment (balloob) [reply src/state-summary/state-card-timer.html]: Ok, I'll remove it.

**Review by andrey-git — APPROVED** [review]
When is it displayed as badge and when as state-card?

**Comment by balloob** [comment]
Displayed as a badge by default. Shown as a state card inside the more info or when added to a group.

**Pull request merged**